harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,840 @@
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to download image {display_name} with {wait_timeout} timed out\n"
+            f"Still got {code} with {data}"
+        )
+
+
+def check_vm_running(api_client, unique_name, wait_timeout):
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+    while endtime > datetime.now():
+        code, data = api_client.vms.get(unique_name)
+        vm_fields = data['metadata']['fields']
+
+        assert 200 == code, (code, data)
+        if vm_fields[2] == 'Running':
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to create VM {unique_name} in Running status, exceed 10 minutes\n"
+            f"Still got {code} with {data}"
+        )
+
+
+def check_vm_ip_exists(api_client, unique_name, wait_timeout):
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+    while endtime > datetime.now():
+        code, data = api_client.vms.vm_instance(unique_name)
+        if 'ipAddress' in data['status']['interfaces'][0]:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to get VM {unique_name} IP address, exceed 10 minutes timed out\n"
+            f"Still got {code} with {data}"
+        )
+
+
+def delete_vm(api_client, unique_name, wait_timeout):
+    api_client.vms.delete(unique_name)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_name)
+        if code == 404:
+            break
+        sleep(5)
+    else:
+        raise AssertionError(
+            f"Failed to delete VM {unique_name}, exceed 10 minutes\n"
+            f"Still got {code} with {data}"
+        )
+
+
+@pytest.mark.p0
+class TestBackendNetwork:
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="mgmt_network_connection")
+    def test_mgmt_network_connection(self, api_client, request, client,
+                                     unique_name, qcow2_image_url):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/validate-network-management-network/
+
+
+        Steps:
+        1. Create a new VM
+        2. Make sure that the network is set to the management network with masquerade as the type
+        3. Wait until the VM boot in running state
+        4. Check can ping VM
+        5. Check can SSH to VM
+        """
+        wait_timeout = request.config.getoption('--wait-timeout')
+        client_ip = request.config.getoption('--endpoint').strip('https://')
+
+        # Check image exists
+        code, data = api_client.images.get(image_name)
+
+        if code == 404:
+            create_image_url(api_client, image_name, qcow2_image_url, wait_timeout)
+
+        # Update AllowTcpForwarding for ssh jumpstart
+        tcp = "sudo sed -i 's/AllowTcpForwarding no/AllowTcpForwarding yes/g' /etc/ssh/sshd_config"
```

Comment:
The command expose the node to be a jumphost, it would better named `jumphost` rather than `tcp`.
And we should always revert the change when leaving.

Or you can simply use `host_shell` fixture in `api_client`.